src/messaging/ExchangeMgr.cpp

```diff
@@ -295,6 +295,19 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
     DispatchMessage(packetHeader, payloadHeader, msgBuf);
 }
 
+void ExchangeManager::OnConnectionExpired(Transport::PeerConnectionState * state, SecureSessionMgrBase * mgr)
+{
+    // Search for an existing exchange that the message applies to. If a match is found...
+    ExchangeContext * ec = ContextPool;
+    for (int i = 0; i < CHIP_CONFIG_MAX_EXCHANGE_CONTEXTS; i++, ec++)
```

Comment:
How about:
```
  for (ExchangeContext & ec : ContextPool)
  {
      if (ec.GetReferenceCount() > 0 && ec.mPeerNodeId == state->GetPeerNodeId())
....
```

Reply:
And this is pre-existing, but shouldn't the member be named mContextPool?

Reply:
This is legacy codes, but I'll be glad to refactor.

Reply:
To be clear, I was not asking you to change all the other loops like that (which I did not even realize existed)!  In the future, it probably makes the most sense to put refactorings that do not change functionality (like changing all the loops) in a separate PR from changes that actually change functionality.  It would definitely make things a lot easier to review....